app/views/submissions/_details.html.erb

```diff
@@ -35,7 +35,7 @@
             <tr>
               <th>Case</th>
               <th>Status</th>
-              <% if !@submission.all_tests_passed? %>
+              <% if !@submission.all_tests_passed? && can?(:read_result, @submission) %>
```

Comment:
Try to be consistent with the use of brackets with `can?`. The bracksts were just removed from app/views/participants/show.html.erb

(I'd prefer it w/o the brackets)